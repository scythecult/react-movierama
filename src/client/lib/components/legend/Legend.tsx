import { Seat } from '../seat/Seat';
import styles from './styles.module.css';

export const Legend = () => {
  return (
    <ul className={styles.legend}>
      <li className={styles.legendItem}>
        <Seat isDisabled />
        <small>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isDisabled isSelected />
        <small>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isOccupied />
        <small>Occupied</small>
      </li>
    </ul>
  );
};
