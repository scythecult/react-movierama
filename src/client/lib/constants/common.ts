export const STORED_MOVIE_KEY = 'movierama_saved_movie';

export const STORED_SEATS_KEY = 'movierama_saved_seats';

export const DEFAULT_CANVAS_WIDTH = 400;

export const DEFAULT_CANVAS_HEIGHT = 400;

export const DEFAULT_TICKET_TYPE_ID = 1;

export const SeatTypeMap = {
  REGULAR: 1,
  COMFORT: 2,
  VIP: 3,
  WHEELCHAIR: 4,
} as const;

export const SeatStateMap = {
  FREE: 1,
  OCCUPIED: 2,
  SELECTED: 3,
} as const;

export const SEAT_TYPE: Record<string, string> = {
  [SeatTypeMap.REGULAR]: 'regular',
  [SeatTypeMap.COMFORT]: 'comfort',
  [SeatTypeMap.VIP]: 'vip',
  [SeatTypeMap.WHEELCHAIR]: 'wheelchair',
} as const;

export const SEAT_STATE: Record<string, string> = {
  [SeatStateMap.FREE]: 'free',
  [SeatStateMap.OCCUPIED]: 'occupied',
  [SeatStateMap.SELECTED]: 'selected',
} as const;
