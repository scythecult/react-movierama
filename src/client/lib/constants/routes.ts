export const AppRoute = {
  ROOT: '/',
  ORDER_PAGE: '/order',
};
