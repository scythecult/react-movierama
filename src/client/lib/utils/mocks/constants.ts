// TODO Should move to server
export const SEAT_WIDTH = 40;

export const SEAT_HEIGHT = 34;

export const SEAT_OFFSET = 10;

export const HALL_PLAN_MATRIX =
  // HALL
  [
    // ROW 1 - seat 0 - empty space width === 1 seat
    [1, 1, 0, 1, 4, 4, 1, 0, 1, 1, 1, 1, 1],
    // ROW
    [2, 2, 0, 1, 1, 1, 1, 0, 1],
    // ROW
    [1, 1, 0, 2, 2, 2, 2, 0, 1],
    // ROW
    [1, 1, 0, 3, 3, 3, 3, 0, 1],
  ];
