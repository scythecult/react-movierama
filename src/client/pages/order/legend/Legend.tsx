import type { SeatType } from '@/client/lib/types/OrderData';
import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export type LegendProps = {
  seatTypes: SeatType[];
};

export const Legend = (props: LegendProps) => {
  const { seatTypes } = props;

  const seatTypeNodes = seatTypes.map((seatType) => {
    const { id, name, ticketTypes } = seatType;
    const [defaultTicketType] = ticketTypes;

    return <li className={styles.legendItem} key={id}>
      <Seat className={styles.legendItemSeat} type={name} />
      <div className={styles.legendItemWrap}>
        <small className={styles.legendItemName}>{name}</small>
        <small className={styles.legendItemPrice}>{defaultTicketType.price} RUB</small>
      </div>
    </li>;
  });

  return (
    <ul className={styles.legend}>
      {seatTypeNodes}
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled />
        <small className={styles.legendItemName}>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled isSelected />
        <small className={styles.legendItemName}>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isOccupied />
        <small className={styles.legendItemName}>Occupied</small>
      </li>
    </ul>
  );
};
