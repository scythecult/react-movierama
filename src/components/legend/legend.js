import React from "react";
import { Seat } from "../seat/seat";

import classes from "./legend.module.css";

const Legend = () => {
  return (
    <ul className={classes.legend}>
      <li className={classes["legend__item"]}>
        <Seat disabled={true} />
        <small>N/A</small>
      </li>
      <li className={classes["legend__item"]}>
        <Seat disabled={true} selected={true} />
        <small>Selected</small>
      </li>
      <li className={classes["legend__item"]}>
        <Seat occupied={true} />
        <small>Occupied</small>
      </li>
    </ul>
  );
};

export { Legend };
