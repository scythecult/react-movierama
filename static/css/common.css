*,
*::before,
*::after {
    box-sizing: border-box;
}

body,
h1,
h2,
h3,
h4,
p,
li,
figure,
figcaption,
blockquote,
dl,
dd {
    margin: 0;
}

html {
    scroll-behavior: smooth;
}

body {
    overflow-y: scroll;
}

img {
    display: block;
    max-width: 100%;
    height: auto;
}

input,
button,
textarea,
select {
    font: inherit;
}

.visually-hidden:not(:focus):not(:active),
input[type='checkbox'].visually-hidden,
input[type='radio'].visually-hidden {
    position: absolute;
    width: 1px;
    height: 1px;
    margin: -1px;
    border: 0;
    padding: 0;
    white-space: nowrap;
    clip-path: inset(100%);
    clip: rect(0 0 0 0);
    overflow: hidden;
}

:root {
    --app-font-family: system-ui, 'Segoe UI', 'Roboto', 'Helvetica', 'Arial', sans-serif, 'Apple Color Emoji',
    'Segoe UI Emoji', 'Segoe UI Symbol';
    --app-primary-bg: #242424;
    --app-secondary-bg: rgba(0, 0, 0, 0.2);
    --app-primary-text: rgba(255, 255, 255, 0.87);
    --app-secondary-text: #777777;
    --app-primary-link: #646cff;
    --app-secondary-link: #535bf2;
    --app-primary-button: #646cff;
    --app-primary-button-hover: #4d55fd;
    --app-secondary-button: #535bf2;
    --app-cyan-color: #6feaf6;
    --app-blue-color: #646cff;
    --app-gray-color: #444451;
    --app-white-color: #ffffff;
    --app-safe-bg: transparent;

    --app-screen-color: #ffffff;
    --app-seat-text: #777777;
    --app-seat-regular-color: #444451;
    --app-seat-comfort-color: #971859;
    --app-seat-vip-color: #ff9a2a;
    --app-seat-wheelchair-color: #ff6347;
    --app-seat-occupied-color: #ffffff;
    --app-seat-selected-color: #6feaf6;

    --app-border-color: #777777;
    --app-border-radius-s: 6px;
    --app-border-radius-m: 10px;
    --app-border-radius-l: 16px;
}

@media (prefers-color-scheme: light) {
    /* слишком яркое изображение дискомфортно для глаз */
    img {
        filter: brightness(90%);
    }

    :root {
        --app-primary-bg: #ffffff;
        --app-secondary-bg: rgba(0, 0, 0, 0.1);
        --app-primary-text: #213547;
        --app-screen-color: #444451;
        --app-safe-bg: rgba(0, 0, 0, 0.1);

        --app-seat-text: #ffffff;
    }
}

body {
    margin: 0;
    min-width: 320px;
    min-height: 100dvh;
    line-height: 1.3;
    font-weight: 400;
    font-family: var(--app-font-family, 'Segoe UI'), sans-serif;
    font-synthesis: none;
    color: var(--app-primary-text);
    background-color: var(--app-primary-bg);
    transition-property: background-color, color;
    transition-duration: 0.2s;
    color-scheme: light dark;
}
