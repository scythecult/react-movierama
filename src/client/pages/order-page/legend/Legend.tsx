import { SeatStateMap } from '@/client/lib/constants/common';
import type { SeatType } from '@/client/lib/types/OrderPageData';
import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export type LegendProps = {
  seatTypes: SeatType[];
};

export const Legend = (props: LegendProps) => {
  const { seatTypes } = props;

  const seatTypeNodes = seatTypes.map((seatType) => {
    const { id, name, ticketTypes } = seatType;
    const [defaultTicketType] = ticketTypes;

    return (
      <li className={styles.legendItem} key={id} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} type={id} />
        <div className={styles.legendItemWrap} data-test-id="legend-item-wrap">
          <small className={styles.legendItemName} data-test-id="legend-item-name">
            {name}
          </small>
          <small className={styles.legendItemPrice} data-test-id="legend-item-price">
            {defaultTicketType.price} RUB
          </small>
        </div>
      </li>
    );
  });

  return (
    <ul className={styles.legend} data-test-id="legend">
      {seatTypeNodes}
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.FREE} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          N/A
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.SELECTED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Selected
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.OCCUPIED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Occupied
        </small>
      </li>
    </ul>
  );
};
