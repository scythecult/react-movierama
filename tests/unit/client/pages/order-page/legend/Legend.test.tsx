import { render } from '@testing-library/react';
import { Legend, type LegendProps } from '@/client/pages/order/legend/Legend';
import { SEAT_TYPES } from '../../../../constants/common';


const DEFAULT_PROPS: LegendProps = {
  seatTypes: [],
};

describe('Legend', () => {
  test('should correspond default layout', () => {
    const result = render(<Legend {...DEFAULT_PROPS} />);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "seatTypes" prop', () => {
    const result = render(<Legend seatTypes={SEAT_TYPES} />);
    expect(result.container).toMatchSnapshot();
  });
});
