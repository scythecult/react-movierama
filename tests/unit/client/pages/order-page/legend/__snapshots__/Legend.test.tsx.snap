// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Legend > should correspond default layout 1`] = `
<div>
  <ul
    class="_legend_28be33"
  >
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        N/A
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _selected_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        Selected
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _occupied_aa3898"
        disabled=""
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        Occupied
      </small>
    </li>
  </ul>
</div>
`;

exports[`Legend > should support the "seatTypes" prop 1`] = `
<div>
  <ul
    class="_legend_28be33"
  >
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_28be33"
      >
        <small
          class="_legendItemName_28be33"
        >
          regular
        </small>
        <small
          class="_legendItemPrice_28be33"
        >
          330
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _comfort_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_28be33"
      >
        <small
          class="_legendItemName_28be33"
        >
          comfort
        </small>
        <small
          class="_legendItemPrice_28be33"
        >
          380
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _vip_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_28be33"
      >
        <small
          class="_legendItemName_28be33"
        >
          vip
        </small>
        <small
          class="_legendItemPrice_28be33"
        >
          430
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _wheelchair_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_28be33"
      >
        <small
          class="_legendItemName_28be33"
        >
          wheelchair
        </small>
        <small
          class="_legendItemPrice_28be33"
        >
          230
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _free_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        N/A
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _selected_aa3898"
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        Selected
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _regular_aa3898 _occupied_aa3898"
        disabled=""
        type="button"
      >
        <span
          class="_seatPlace_aa3898"
        />
        <svg
          class="_seatIcon_aa3898"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_28be33"
      >
        Occupied
      </small>
    </li>
  </ul>
</div>
`;
